src/controllers/hooks/router-store/index.tsx

```diff
@@ -18,3 +20,36 @@ export const useRouterActions = (): RouterActionsType => {
 
   return allActions;
 };
+
+/**
+ * Utility hook for accessing URL query params
+ */
+export const useQueryParam = (
+  paramKey: string
+): [string | undefined, (qp: string | null) => void] => {
+  const [entireState, routerActions] = useRouterStore();
```

Comment:
I think this will always re-render. We need to create a dedicated hook with a selector to avoid that.

Reply:
Nice catch. I re worked on the logic using a selector. Force pushed the code.